docs/E2ETESTING.md

```diff
@@ -0,0 +1,77 @@
+# End-to-End (E2E) Testing with Stark
+
+The stark starter comes with a setup to run e2e tests with protractor.
+
+The showcase app uses Travis CI and BrowserStack to automatically run e2e-tests on multiple setups (IE, Safari, Android, IOS, ...).
+
+## Protractor
+
+Protractor is an end-to-end test framework for Angular and AngularJS applications. Protractor runs tests against your application running in a real browser, interacting with it as a user would.
+
+For more information go to [https://www.protractortest.org](https://www.protractortest.org).
+
+## BrowserStack (optional)
+
+If you want to run your e2e-tests automatically on multiple setups one of the easiest solutions is BrowserStack.
+They offer a free solution for open source projects.
+
+For more information go to [https://www.browserstack.com/automate](https://www.browserstack.com/automate).
+
+## Running end-to-end tests
+
+Install all dependencies.
+
+```shell
+npm run install:all
+```
+
+Build the application
+
+```shell
+npm run build:showcase
+```
+
+Run the end-to-end tests with protractor.
+
+```shell
+npm run test:showcase:e2e
+```
+
+## Running end-to-end tests with BrowserStack
+
+### Prerequisites
+
+-   BrowserStack Account ([https://www.browserstack.com/](https://www.browserstack.com/))
+
+### Setup
+
+Find your credentials on the [BrowserStack dashboard](https://automate.browserstack.com/dashboard).
+
+Set the environmental `BROWSERSTACK_USERNAME` to the username and `BROWSERSTACK_ACCESS_KEY` to the access key.
+
+### Run the tests
+
+```shell
+npm run test:e2e:showcase:browserstack
+```
+
+_If you want to change the devices / browsers on which the tests are run you can change the `CAPABILITIES` variable in `showcase/protractor.browserstack.conf.js`. For the correct configurations see: [https://www.browserstack.com/automate/capabilities](https://www.browserstack.com/automate/capabilities)_
```

Comment:
Maybe we can re-phrase the second part like this:
"... you can change the `CAPABILITIES` variable in the protractor config file. You can have a look on  `showcase/protractor.browserstack.conf.js` to see in which devices the Stark Showcase is tested"

Reply:
### Run the tests

```shell
npm run test:e2e:showcase:browserstack
```

_If you want to integrate BrowserStack in your own project you can have a look at `showcase/e2e/protractor.browserstack.conf.js` where we configured our setup._

_For more documentation see:_ 
  - _[https://github.com/angular/protractor/blob/master/lib/config.ts](https://github.com/angular/protractor/blob/master/lib/config.ts)_
  - _[https://www.browserstack.com/automate/capabilities](https://www.browserstack.com/automate/capabilities)_

Reply:
How is that?